bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitCharacteristicFactory.java

```diff
@@ -200,8 +202,8 @@ private static void setStateFromEnum(GenericItem item, CharacteristicEnum value,
     }
 
     public static Supplier<CompletableFuture<Double>> getCompletedFutureDouble(Item item) {
-        return () -> CompletableFuture.completedFuture(
-                item.getStateAs(DecimalType.class) != null ? item.getStateAs(DecimalType.class).doubleValue() : 0.0);
+        final DecimalType value = item.getStateAs(DecimalType.class);
```

Comment:
@J-N-K question here. it returns a supplier that gets executed when HomeKit retrieves the value of an item. with your change, will it still call item.getStateAs or will it always return the same double value?